File: web-app/online-warehouse/src/app/counterparty/dto/create-counterparty.dto.ts
import {AddressDto} from "../../shared/address/dto/address.dto";
import {CounterpartyTypeEnum} from "./enum/counterparty-type.enum";

export class CreateCounterpartyDto {
  address: AddressDto;
  companyId: number;
  name: string;
  counterpartyType: CounterpartyTypeEnum;
  taxNumber: string;


  constructor(address?: AddressDto, companyId?: number, name?: string, counterpartyType?: CounterpartyTypeEnum, taxNumber?: string) {
    this.address = address;
    this.companyId = companyId;
    this.name = name;
    this.counterpartyType = counterpartyType;
    this.taxNumber = taxNumber;
  }
}

File: core/src/main/java/by/itechart/counterparty/dto/CreateCounterpartyDto.java
package by.itechart.counterparty.dto;

import by.itechart.common.dto.AddressDto;
import by.itechart.counterparty.enums.CounterpartyType;
import lombok.Data;

import javax.validation.constraints.NotEmpty;
import javax.validation.constraints.NotNull;

@Data
public class CreateCounterpartyDto {
    @NotNull
    private AddressDto address;
    @NotNull
    private Long companyId;
    @NotEmpty
    private String name;
    @NotNull
    private CounterpartyType counterpartyType;
    @NotEmpty
    private String taxNumber;
}
